sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADOAuth2AutoConfiguration.java

```diff
@@ -36,29 +42,69 @@
 @ConditionalOnResource(resources = "classpath:aad.enable.config")
 @ConditionalOnWebApplication(type = ConditionalOnWebApplication.Type.SERVLET)
 @ConditionalOnProperty(prefix = "azure.activedirectory", value = "tenant-id")
-@PropertySource(value = "classpath:aad-oauth2-common.properties")
 @PropertySource(value = "classpath:service-endpoints.properties")
 @EnableConfigurationProperties({ AADAuthenticationProperties.class, ServiceEndpointsProperties.class })
 public class AADOAuth2AutoConfiguration {
 
-    private final AADAuthenticationProperties aadAuthProps;
-    private final ServiceEndpointsProperties serviceEndpointsProps;
+    private final AADAuthenticationProperties aadAuthenticationProperties;
+    private final ServiceEndpointsProperties serviceEndpointsProperties;
 
     public AADOAuth2AutoConfiguration(AADAuthenticationProperties aadAuthProperties,
-                                      ServiceEndpointsProperties serviceEndpointsProps) {
-        this.aadAuthProps = aadAuthProperties;
-        this.serviceEndpointsProps = serviceEndpointsProps;
+                                      ServiceEndpointsProperties serviceEndpointsProperties) {
+        this.aadAuthenticationProperties = aadAuthProperties;
+        this.serviceEndpointsProperties = serviceEndpointsProperties;
     }
 
     @Bean
     @ConditionalOnProperty(prefix = "azure.activedirectory.user-group", value = "allowed-groups")
     public OAuth2UserService<OidcUserRequest, OidcUser> oidcUserService() {
-        return new AADOAuth2UserService(aadAuthProps, serviceEndpointsProps);
+        return new AADOAuth2UserService(aadAuthenticationProperties, serviceEndpointsProperties);
+    }
+
+    @Bean
+    public ClientRegistrationRepository clientRegistrationRepository() {
+        return new InMemoryClientRegistrationRepository(azureClientRegistration());
+    }
+
+    private ClientRegistration azureClientRegistration() {
+        String tenantId = aadAuthenticationProperties.getTenantId().trim();
+        Assert.hasText(tenantId, "azure.activedirectory.tenant-id should have text.");
+        Assert.doesNotContain(tenantId, " ", "azure.activedirectory.tenant-id should not contain ' '.");
+        Assert.doesNotContain(tenantId, "/", "azure.activedirectory.tenant-id should not contain '/'.");
+        return ClientRegistration.withRegistrationId("azure")
```

Comment:
Will multi-tenant application still function with this change?

Reply:
IMU, if we set `azure.activedirectory.tenant-id=common`, then multi-tenant application will function.